### ISSUE TITLE
Use of multiple BigInt implementations

### ISSUE THREAD

**Issue by SamuelMarks** [Fantom-foundation/evm-rs#1]
You use both [primitive-types](https://crates.io/crates/primitive-types) and [etcommon-bigint](https://docs.rs/etcommon-bigint) which causes issues here:
https://github.com/Fantom-foundation/evm-rs/blob/cdd3f43/fvm/src/vm.rs#L289
```rust
self.registers[self.stack_pointer - 1] = M256::from(&*k);
```

There are two mutually incompatible definitions of the same type:
```rust
construct_fixed_hash! {
	/// Fixed-size uninterpreted hash type with 32 bytes (256 bits) size.
	pub struct H256(32);
}
```

```rust
impl_hash!(H256, 32);
```

Maybe we could add `From` for every type, and potentially get it to work without copying data, but I doubt that would be a good solution.

Alternatives:
- Move away from [keccak-hash](https://github.com/paritytech/parity-common/tree/master/keccak-hash)
- Move away from etcommon-bigint; use primitive-types instead

What do you think?

**Comment by AgustinCB** [comment]
The problem is that `primitive-types` is used by `keccak-hash` but doesn't implement `Serialize`, and `etcommon-bigint` implements `Serialize` but is not used by `keccak-hash`.

If there's a library for Keccak that uses `etcommon-bigint`, happy to migrate.

**Comment by SamuelMarks** [comment]
Closed by #2
